src/main/java/hudson/plugins/favorite/Favorites.java

```diff
@@ -150,7 +152,7 @@ public void remove() {
      * @throws IllegalArgumentException when the user is anonymous
      */
     private static FavoriteUserProperty getProperty(@NonNull User user) throws IOException {
-        if (FavoritePlugin.isAnonymous(user)) {
+        if ("anonymous".equalsIgnoreCase(user.getFullName())) {
```

Comment:
That is wrong. If `user` is not null, it means that the user is logged in and is not anonymous.

Reply:
You may be right, I was just keeping the logic that was there before. If that is indeed the case then there is no need for the anonymous check when user is declared as `@NonNull`. I can clean that up in a future PR.